openeo/extra/job_management.py

```diff
@@ -273,6 +277,36 @@ def _normalize_df(self, df: pd.DataFrame) -> pd.DataFrame:
 
         return df
 
+    def start_job_thread(self,start_job: Callable[[], BatchJob],
+        job_db: JobDatabaseInterface ):
+        """
+        Start running the jobs in a separate thread, returns afterwards.
+        """
+
+        # Resume from existing db
+        _log.info(f"Resuming `run_jobs` from existing {job_db}")
+        df = job_db.read()
+
+        self._stop = False
+
+
+
+        def run_loop():
+            while (sum(job_db.count_by_status(statuses=["not_started","created","queued","running"]).values()) > 0 and not self._stop
+            ):
+                self._job_update_loop(df, job_db, start_job)
+
+
+
+        self._timer = Thread(target = run_loop)
+        self._timer.start()
+
+    def stop_job_thread(self, force_timeout_seconds = 30):
+        self._stop = True
+        if(self._timer is not None):
+            self._timer.join(force_timeout_seconds)
```

Comment:
This "force" here is misleading I think, It won't force the killing of the thread, it's just a timeout to wait for it 
the return here won't be a guarantee that the thread has finished.
At least I would print a warning when the thread is still alive (`.is_alive()`) before returning